### ISSUE TITLE
Authentik with OIDC not working

### ISSUE THREAD

**Issue by saschabrockel** [dominik-th/matomo-plugin-LoginOIDC#114]
I configured everything which was a problem anyway and now I always get the following error out of nowhere when being redirected to Matomo:

An error occurred
Unexpected response from the OAuth service.

There are no detailed logs about this.

My plugin settings:
```
Authorize URL: https://auth.mydomain.tld/login/oauth/authorize
Token URL: https://auth.mydomain.tld/login/oauth/access_token
Userinfo URL: https://auth.mydomain.tld/user
Logout URL: empty
Userinfo ID: sub
Client ID: ...
Client Secret: ...
OAuth Scopes: openid email profile
```
I have set to true:
- disable direct login link
- create automatically a new account when a unknown user logs in
- disable second factor when sign in with OIDC

In Authentik it is set (had to escape the ? in the URL):
```
Flow: implicit-consent
Client ID: ...
Client Secret: ...
Redirect URI: https://stats.mydomain.tld/index.php\?module=LoginOIDC&action=callback&provider=oidc
Signing Key: authentic Self-signed Cert
Scopes: email openid profile
Subject mode: Based on the User's hashed ID
```

I click on the button in Matomo then get redirected to Authentik and can also login and then get redirected to this URL with error described above: https://stats.mydomain.tld/index.php?module=LoginOIDC&action=callback&provider=oidc&code=0a66582e56fa4c9e8d9da91af52613da&state=92f10201e9982602cc071764c0de3359

**Comment by luxuskommunismus** [comment]
Same Problem, I have also tried to set the redirect URI to .* as suggested here: https://github.com/dominik-th/matomo-plugin-LoginOIDC/pull/86

The error is triggered in line 235 and 258 of controller.php (https://github.com/dominik-th/matomo-plugin-LoginOIDC/blob/5.x-dev/Controller.php) but I have no idea of the cause. 

Let me know if you manage to fix this, I think this repository is mostly abandoned

**Comment by luxuskommunismus** [comment]
Just to add: Authentik logs an "invalid client secret-error" for me but what that means is beyond my knowledge.  {"auth_via": "oauth_client_secret", "client_id": "****", "domain_url": "*****", "event": "Invalid client secret", "host": "*****", "level": "warning"}

**Comment by andreiwow2** [comment]
I have the same issue

**Comment by trailfog** [comment]
Authentik uses a regex to define callback URLs. Authentik seems to auto escape the forward slash, however the `?` is not automatically escaped.

When I set the Redirect URL to `https://auth.mydomain.tld/index.php\?module=LoginOIDC&action=callback&provider=oidc` (take notice of `\?`), everything works as expected.

**Comment by Oliver3911** [comment]
sadly non of these options worked for me

I tried the Redirect URL in RegEx mode with 
`https://matomo.example.org/index.php\?module=LoginOIDC&action=callback&provider=oidc`
in Strict mode with
`https://matomo.example.org/index.php?module=LoginOIDC&action=callback&provider=oidc`
and the whole whitelist `.*`

Login Page forwarding to Authentik and login works fine. Once Authentik redirects me back I get the matomo Page with the 
```
An error occurred
Unexpected response from OAuth service.
```
Error
The page URL is
`https://matomo.example.org/index.php?module=LoginOIDC&action=callback&provider=oidc&code=*REMOVED*&state=*REMOVED*`

Sadly nothing more in Logs which I could find.

Maybe I should looking at the LDAP plugin with Authentik

**Comment by trailfog** [comment]
You need the following settings for Authentik to work:

Matomo:
- Authorize URL: `https://<authentik-domain>/application/o/authorize/`
- Token URL: `https://<authentik-domain>/application/o/token/`
- Userinfo URL: `https://<authentik-domain>/application/o/userinfo/`
- Logout URL: `https://<authentik-domain>/application/o/matomo/end-session/`
- Userinfo ID: `sub`
- Client ID: *Use the client ID generated for you*
- Client Secret: *Use the client secret generated for you*
- OAuth Scopes: `openid email profile`
- Redirect URI override: *Leave this empty*
- Restrict user creation to domains: *to your choosing*

Authentik:
- Redirect URI (Strict): `https://<matomo-domain>/index.php?module=LoginOIDC&action=callback&provider=oidc`

Since my last comment, Authentik allows you to configure if the redirect URL is strict or Regex. Using strict should be used in this case, eliminating the need to escape the `?`.

I also just set up a new Matomo instance, and it worked just fine with the above settings.

@Oliver3911 The Redirect URI is not the issue in your case. Either the `Userinfo ID` or `OAuth Scopes` are probably set incorrectly. When Matomo reports your error, the Redirect URI check has already passed.

**Comment by saschabrockel** [comment]
@trailfog yes this works. Please note the trailing slash. You need it. Otherwise, Authentik will say not found. I will close the issue.

**Comment by e36Alex** [comment]
I got the same problem, even after checking all your advises:

An error occurred
Unerwartete Antwort vom OAuth-Service.

URL: https://domain.de/analytics/index.php?module=LoginOIDC&action=callback&provider=oidc&code=*removed*&state=*removed*
